src/plugins/intel_npu/tests/functional/behavior/npuw/behavior_tests.cpp

```diff
@@ -3,29 +3,37 @@
 //
 
 #include "behavior_tests.hpp"
-#include "openvino/opsets/opset11.hpp"
-#include "openvino/runtime/exec_model_info.hpp"
-#include "openvino/runtime/internal_properties.hpp"
-#include "openvino/runtime/make_tensor.hpp"
-#include "openvino/runtime/properties.hpp"
-#include "openvino/util/file_util.hpp"
 #include "openvino/util/shared_object.hpp"
-#include "npuw_private_properties.hpp"
+#include "compiled_model.hpp"
 
 using namespace testing;
 using namespace ov::npuw::tests;
-
-std::shared_ptr<ov::Model> BehaviorTestsNPUW::create_example_model() {
-    auto param = std::make_shared<ov::opset11::Parameter>(ov::element::i64, ov::PartialShape{1, 3, 2, 2});
-    param->set_friendly_name("input");
-    auto const_value = ov::opset11::Constant::create(ov::element::i64, ov::Shape{1, 1, 1, 1}, {1});
-    const_value->set_friendly_name("const_val");
-    auto add = std::make_shared<ov::opset11::Add>(param, const_value);
-    add->set_friendly_name("add");
-    auto result = std::make_shared<ov::opset11::Result>(add);
-    result->set_friendly_name("res");
-    return std::make_shared<ov::Model>(ov::ResultVector{result}, ov::ParameterVector{param});
-}
+using namespace ov::intel_npu::npuw;
+
+#define THROW(...) \
+    .WillOnce(Throw(std::runtime_error(__VA_ARGS__)))
+
+#define EXPECT_COMPILE_MODEL(device, times, ...)                        \
+    EXPECT_CALL(*device##_plugin,                                       \
+        compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)) \
+        .Times(times)                                                   \
+        __VA_ARGS__                                                     \
+
+#define MODEL(idx) idx
```

Comment:
Why do you need this?

Reply:
To differentiate in macro `EXPECT_CREATE_SYNC_INFER_REQ(mock_npu, MODEL(0), 1)` two last arguments: one for model idx and another one for number of times we expect this call will be done